Add tab completion for Export-Counter -FileFormat parameter

MiaRomero opened the pull request:
* Fixing #3180 
* Added ValidateSet to the -FileFormat parameter for possible log file types and updated the function that was previously validating the file formats.
* Removed the invalid format error message from resources files.
* Updated tests to reflect new FullyQualifiedErrorId.

Commit:
add tab completion for -FileFormat parameter

Commit:
update expected error for invalid -FileFormat test case

iSazonov commented on src/Microsoft.PowerShell.Commands.Diagnostics/ExportCounterCommand.cs:
```diff
@@ -65,12 +65,13 @@ public string Path
                 ValueFromPipelineByPropertyName = false,
                 HelpMessageBaseName = "GetEventResources")]
         [ValidateNotNull]
+        [ValidateSet("blg", "csv", "tsv", IgnoreCase = true)]
```
We can remove `IgnoreCase = true` - it is by default.

MiaRomero replied:
Fixed.

MiaRomero commented on src/Microsoft.PowerShell.Commands.Diagnostics/ExportCounterCommand.cs:
```diff
@@ -180,9 +181,9 @@ protected override void BeginProcessing()
 
             //
-            // Validate the Format and CounterSamples arguments
+            // Validate output format (log file type) and CounterSamples arguments
```
Fixed.

adityapatwardhan commented on src/Microsoft.PowerShell.Commands.Diagnostics/ExportCounterCommand.cs:
```diff
@@ -311,26 +312,21 @@ protected override void ProcessRecord()
 
-        // ValidateFormat() helper.
-        // Validates Format argument: only "BLG", "TSV" and "CSV" are valid strings (case-insensitive)
+        // SetOutputFormat() helper.
```
This line seems unnecessary.

MiaRomero replied:
Removed line 314

adityapatwardhan commented on src/Microsoft.PowerShell.Commands.Diagnostics/ExportCounterCommand.cs:
```diff
@@ -180,9 +181,9 @@ protected override void BeginProcessing()
 
             //
-            // Validate the Format and CounterSamples arguments
+            // Set output format (log file type) and validate CounterSamples arguments
```
Please update comment, SetOuputFormat does not do any validation anymore.

MiaRomero replied:
Updated comment

adityapatwardhan commented on src/Microsoft.PowerShell.Commands.Diagnostics/ExportCounterCommand.cs:
```diff
@@ -65,12 +65,13 @@ public string Path
                 ValueFromPipelineByPropertyName = false,
                 HelpMessageBaseName = "GetEventResources")]
         [ValidateNotNull]
+        [ValidateSet("blg", "csv", "tsv")]
```
Had a chat with Jim. It is ok to keep them as is. Please open an issue on https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/5.1/Microsoft.PowerShell.Diagnostics/Export-Counter.md to explain what they mean.

MiaRomero replied:
The file types are explained in the 'Description' section of the documentation.  Did you want them in the 'Parameter' section as well?

adityapatwardhan replied:
@MiaRomero Yes, please add them to the parameters section as well.

MiaRomero replied:
I've opened issue https://github.com/PowerShell/PowerShell-Docs/issues/1245 for this

SteveL-MSFT reviewed (changes requested):
please add a test case to TabCompletion.Tests.ps1

MiaRomero commented on test/powershell/Modules/Microsoft.PowerShell.Diagnostics/Export-Counter.Tests.ps1:
```diff
@@ -180,7 +180,7 @@ Describe "Feature tests for Export-Counter cmdlet" -Tags "Feature" {
             @{
```
Removed test case

MiaRomero commented:
Hi @TravisEz13,
I've addressed @adityapatwardhan's and @SteveL-MSFT's comments regarding the cmdlet and test code.  I added a comment asking for clarification on the documentation.  Once I have that I can open the appropriate issue.

lzybkr commented on test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1:
```diff
@@ -103,4 +103,10 @@ Describe "TabCompletion" -Tags CI {
+    It 'Should complete "Export-Counter -FileFormat" with available output formats' {
+        $res = TabExpansion2 -inputScript 'Export-Counter -FileFormat ' -cursorColumn 'Export-Counter -FileFormat '.Length
+        $res.CompletionMatches.Count | Should Be 3
+        $res.CompletionMatches.Foreach{$_.CompletionText -in 'blg', 'csv', 'tsv' | Should Be $true}
```
We try to avoid `Should Be $true` because it is difficult to understand what went wrong just reading the logs.

In this specific case, I would instead use:

```powershell
$rest.CompletionMatches.CompletionText -join ' ' | Should Be 'blg csv tsv'
```

iSazonov replied:
There might be a problem with the element's order.
We can use [the auxiliary output](https://github.com/PowerShell/PowerShell/blob/master/test/powershell/engine/DefaultCommands.Tests.ps1#L488)

lzybkr replied:
If order is a concern, pipe to sort first. But in this case, if order is not preserved, I would call that a bug.

MiaRomero replied:
@lzybkr , thank you, I've made the change.  Is it appropriate to go ahead and fix the other test cases in this file that use the same pattern?  Or is that a separate issue/PR?

lzybkr replied:
If you see the pattern in other files, then maybe a new PR, otherwise this PR is fine.

MiaRomero commented:
@TravisEz13 thanks for the link, updated my profile.
